src/pydvl/value/shapley/classwise.py

```diff
@@ -0,0 +1,301 @@
+    :param eps: The threshold when updating using the truncated monte carlo estimator.
+    :param num_samples: The number of subset samples to be drawn from the complement set.
+    :return: The updated result object.
+    """
+    if active_indices is None:
+        active_indices = u.data.indices
+
+    _, y_train = u.data.get_training_data(active_indices)
+    label_set, complement_label_set = __split_into_label_sets(
+        y_train, label, active_indices
+    )
+    _, complement_y_train = u.data.get_training_data(complement_label_set)
+
+    for subset_complement in random_powerset_group_conditional(
+        complement_label_set,
+        complement_y_train,
+        n_samples=num_samples,
+    ):
+        result = __truncated_permutation_mcmc_sampler(
+            u, np.random.permutation(label_set), subset_complement, result, atol=eps
+        )
+
+    return result
+
+
+def _estimate_in_out_cls_accuracy(
+    model: SupervisedModel, x: np.ndarray, labels: np.ndarray, label: np.int_
+) -> Tuple[float, float]:
+    """
+    Estimate the in and out of class accuracy as defined in [1], Equation 3.
+
+    :param model: A model to be used for predicting the labels.
+    :param x: The inputs to be used for measuring the accuracies. Has to match the labels.
+    :param labels: The labels ot be used for measuring the accuracies. It is divided further by the passed label.
+    :param label: The label of the class, which is currently viewed.
+    :return: A tuple, containing the in class accuracy as well as the out of class accuracy.
+    """
+    n = len(x)
+    y_pred = model.predict(x)
+    label_set_match = labels == label
+    label_set = np.where(label_set_match)[0]
+    complement_label_set = np.where(~label_set_match)[0]
+
+    acc_in_cls = accuracy_score(labels[label_set], y_pred[label_set], normalize=False)
+    acc_out_of_cls = accuracy_score(
+        labels[complement_label_set], y_pred[complement_label_set], normalize=False
+    )
+    return acc_in_cls / n, acc_out_of_cls / n
+
+
+def __split_into_label_sets(
+    labels: IntArray, label: int, indices: IntArray
+) -> Tuple[IntArray, IntArray]:
+    """
+    Splits the indices into two sets, one containing the indices with the label and the other the remaining indices.
+    :param labels: The labels of the indices.
+    :param label: The label to be used for splitting.
+    :param indices: The indices to be split.
+    :return: The two sets of indices.
+    """
+    active_elements = labels == label
+    label_set = np.where(active_elements)[0]
+    complement_label_set = np.where(~active_elements)[0]
+    label_set = indices[label_set]
+    complement_label_set = indices[complement_label_set]
+    return label_set, complement_label_set
+
+
+def __truncated_permutation_mcmc_sampler(
+    u: Utility,
+    class_label_set: np.ndarray,
+    complement_set: np.ndarray,
+    result: ValuationResult,
+    atol: float,
+) -> ValuationResult:
```

Comment:
As before:
* this is not a "sampler"
* better not to use double underscores
* Use NDArray
* Don't use tolerances, but `StoppingCriterion` or `TruncationPolicy`